src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/CoffObjectWriter.cs

```diff
@@ -0,0 +1,1097 @@
+                        // associative section, so we are stuck with one section per each
+                        // unwind symbol.
+                        xdataSectionWriter = GetOrCreateSection(ObjectNodeSection.XDataSection, currentSymbolName, unwindSymbolName);
+                        pdataSectionWriter = GetOrCreateSection(PDataSection, currentSymbolName, null);
+                    }
+                    else
+                    {
+                        xdataSectionWriter = _xdataSectionWriter;
+                        pdataSectionWriter = _pdataSectionWriter;
+                    }
+
+                    // Need to emit the UNWIND_INFO at 4-byte alignment to ensure that the
+                    // pointer has the lower two bits in .pdata section set to zero. On ARM64
+                    // non-zero bits would mean a compact encoding.
+                    xdataSectionWriter.EmitAlignment(4);
+
+                    xdataSectionWriter.EmitSymbolDefinition(unwindSymbolName);
+
+                    // Emit UNWIND_INFO
+                    xdataSectionWriter.Write(blob);
+
+                    FrameInfoFlags flags = frameInfo.Flags;
+
+                    if (i != 0)
+                    {
+                        xdataSectionWriter.WriteByte((byte)flags);
+                    }
+                    else
+                    {
+                        MethodExceptionHandlingInfoNode ehInfo = nodeWithCodeInfo.EHInfo;
+                        ISymbolNode associatedDataNode = nodeWithCodeInfo.GetAssociatedDataNode(_nodeFactory) as ISymbolNode;
+
+                        flags |= ehInfo is not null ? FrameInfoFlags.HasEHInfo : 0;
+                        flags |= associatedDataNode is not null ? FrameInfoFlags.HasAssociatedData : 0;
+
+                        xdataSectionWriter.WriteByte((byte)flags);
+
+                        if (associatedDataNode is not null)
+                        {
+                            xdataSectionWriter.EmitSymbolReference(
+                                IMAGE_REL_BASED_ADDR32NB,
+                                GetMangledName(associatedDataNode));
+                        }
+
+                        if (ehInfo is not null)
+                        {
+                            xdataSectionWriter.EmitSymbolReference(
+                                IMAGE_REL_BASED_ADDR32NB,
+                                GetMangledName(ehInfo));
+                        }
+
+                        if (nodeWithCodeInfo.GCInfo is not null)
+                        {
+                            xdataSectionWriter.Write(nodeWithCodeInfo.GCInfo);
+                        }
+                    }
+
+                    // Emit RUNTIME_FUNCTION
+                    pdataSectionWriter.EmitAlignment(4);
+                    pdataSectionWriter.EmitSymbolReference(IMAGE_REL_BASED_ADDR32NB, currentSymbolName, start);
+                    // Only x64 has the End symbol
+                    if (_machine == Machine.Amd64)
```

Comment:
Just wondering - with the old ObjectWriter would we ever emit packed unwind info on ARM64?

Reply:
No, we would not. It makes sense to open an issue to track whether we can do that. There’s already an issue for macOS compact unwinding and I assume the potential set of problems will be similar (JIT ABI matching the format, unwinding when in prolog).